analyzer/src/main/kotlin/managers/GoMod.kt

```diff
@@ -514,9 +514,9 @@ private data class ModuleInfoFile(
     )
 }
 
-private fun GoMod.ModuleInfo.toVcsInfo(): VcsInfo {
+private fun GoMod.ModuleInfo.toVcsInfo(): VcsInfo? {
     val info = jsonMapper.readValue<ModuleInfoFile>(File(goMod).resolveSibling("$version.info"))
-    val type = info.origin.vcs?.let { VcsType.forName(it) }.takeIf { it == VcsType.GIT } ?: return VcsInfo.EMPTY
+    val type = info.origin.vcs?.let { VcsType.forName(it) }.takeIf { it == VcsType.GIT } ?: return null
```

Comment:
This reminds me that we should probably rather say `.takeUnless { it.type == VcsType.UNKNOWN }` here, either as part of this commit, or even already as part of the commit that introduced `.takeIf { it == VcsType.GIT }`.

Reply:
Also, that would allow you to keep `?: return VcsInfo.EMPTY` here, but in the caller just use `toVcsInfo()` (without `orEmpty()`).

Reply:
By using `.takeUnless { it.type == VcsType.UNKNOWN }` this would extend the code beyond just `git` as VCS.
This would be completely untested. I neither would like to extend the scope of this commit to test this nor I'd like to do this fully untested.

Reply:
I'm still wondering why you're not leaving `toVcsInfo()` untouched and instead just change line 312 to `toVcsInfo()`, because `toVcsInfo()` can never return `VcsType.UNKNOWN`.

Reply:
I find my current version cleaner, `toVcsInfo()` return non-null if and only if it was able to determine VCS info.